.github/workflows/tests.yaml

```diff
@@ -1,36 +1,50 @@
 name: Build
-on: [push]
+
+on:
+  push:
+    branches: [ main ]
+  pull_request:
+    branches: [ main ]
+  release:
+    types: [ published ]
+  workflow_dispatch:
+
 jobs:
-  no-libvarnish:
+  test:
     runs-on: ubuntu-latest
+    env:
+      RUST_BACKTRACE: 1
     steps:
       - uses: actions/checkout@v4
-      - uses: actions-rs/toolchain@v1
-        with:
-          toolchain: stable
-      - run: |
-          cargo doc
-  with-libvarnish:
+      - name: rust-info
+        run: rustc --version && cargo --version
+      - uses: Swatinem/rust-cache@v2
+        if: github.event_name != 'release' && github.event_name != 'workflow_dispatch'
+      - name: install varnish-dev
+        run: |
+          curl -s https://packagecloud.io/install/repositories/varnishcache/varnish75/script.deb.sh | sudo bash
+          sudo apt-get install -y varnish-dev
+      - name: test-fmt
+        run: cargo fmt --all -- --check
+      - name: check
+        run: RUSTFLAGS='-D warnings' cargo check --workspace --all-targets
+      - name: build vmod
+        run: cargo build --all-targets
```

Comment:
you need to build all the workspace as the `cargo test` below will require the vmods to be built

Reply:
i was a bit confused by that - test (in theory) must build the code. How is it possible for the test to run without building it first?

Reply:
hmm, i guess in theory `test` might skip the step in which build places final binaries to the `release|debug` dir... Ok, will have to simply modify check -> build.